ShoppingList/src/main/java/edu/agh/eaiib/model/Product.java

```diff
@@ -21,18 +23,14 @@ public int getAmount() {
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
-
         Product product = (Product) o;
-
-        if (amount != product.amount) return false;
-        return name != null ? name.equals(product.name) : product.name == null;
+        return amount == product.amount &&
+                Objects.equals(name, product.name);
     }
 
     @Override
     public int hashCode() {
-        int result = name != null ? name.hashCode() : 0;
-        result = 31 * result + amount;
-        return result;
+        return Objects.hash(name, amount);
```

Comment:
Nice, I didn't know this method 🙈

Reply:
It is also used when you automatically add hashCode in IntelliJ with Alt+Insert.

Reply:
My IntellIJ generated this deleted code.